### PR TITLE
Added support for the NXP K64 series

### PR DESCRIPTION
**Pull request by emb4fun**
Here I am working with a FRDM-K64F board and try to get the TinyUSB up and running.
It takes some time, but I could found the problem why it is not working at first.

In case of the K64 the USB clock will be set with e.g.

`CLOCK_EnableUsbfs0Clock(kCLOCK_UsbSrcIrc48M, 48000000U);`

This function will set inside some registers which are needed for the USB clock recovery:

```
if (kCLOCK_UsbSrcIrc48M == src)
{
    USB0->CLK_RECOVER_IRC_EN = 0x03U;
    USB0->CLK_RECOVER_CTRL |= USB_CLK_RECOVER_CTRL_CLOCK_RECOVER_EN_MASK;
}
```

Unfortunately this registers will be reset by the dcd_init function where a USB reset is performed.
This PR fix the problem in case of a K64 series. With this changes the device will be enumerated.
I am still in the progress of testing.

Best regards,
Michael

### CODE REVIEW

**Review by hathach — COMMENTED** [review]
thank you very much for your PR. I have rename the option from `OPT_MCU_KINETIS_K64` to simply `OPT_MCU_KINETIS_K` since it seems to be the name of family of those m4f.

I have tried to add kinetis_k to bsp with teensy_35 board to test with this PR. Unfortunately teensy 3.5 is not able to blink led with board_test example. It is probably due to clock config. I push the changes anyway, will try to find an k64f board later to test with. Menanwhile please be patient.

File: hw/bsp/kinetis_k/family.mk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+MCU_DIR = $(SDK_DIR)/devices/${MCU_VARIANT}
+include $(TOP)/$(BOARD_PATH)/board.mk
+CPU_CORE ?= cortex-m4
+
+CFLAGS += \
+  -DCFG_TUSB_MCU=OPT_MCU_KINETIS_K \
+
+LDFLAGS += \
+  -Wl,--defsym,__stack_size__=0x400 \
+  -Wl,--defsym,__heap_size__=0
```
Comment (karlp) [inline hw/bsp/kinetis_k/family.mk:13]: why did you need to specify these explicitly?

File: src/portable/nxp/khci/dcd_khci.c [hunk 2]
```diff
@@ -269,9 +269,23 @@ void dcd_init(uint8_t rhport)
...
   KHCI->USBTRC0 |= USB_USBTRC0_USBRESET_MASK;
   while (KHCI->USBTRC0 & USB_USBTRC0_USBRESET_MASK);
 
+  KHCI->CLK_RECOVER_IRC_EN = clk_recover_irc_en;
+  KHCI->CLK_RECOVER_CTRL  |= clk_recover_ctrl;
+#else
+  KHCI->USBTRC0 |= USB_USBTRC0_USBRESET_MASK;
+  while (KHCI->USBTRC0 & USB_USBTRC0_USBRESET_MASK);
+#endif
+
```
[the remark this answers is not in the document]
Comment (hathach) [reply src/portable/nxp/khci/dcd_khci.c:288]: I think it is rather legit, we should reserve these value for crystal-less USB.

**Comment by hathach** [comment]
> Why did you need the freertosconfig.h file? That seems unnecessary to have another copy of it? (and freertos demos work fine for me without it)
> 
> I'm still clearing our work with internal management, so it's not (very) publically shared yet unfortunately.

It is part of my new approach, previously there is 1 freertos (per example) for all ports. However, that makes it difficult to compile with IAR since freeRTOSConfig.h is included by IAR assembler which is very primitive and cannot support complicated include like gcc assembler. SO I have moved freeRTOSconfig.h to each bsp, it kind of make sense as well, since each hardware would have different freertosconfig. 

Currently only cmake make use of the new config (make still use the example's one), but eventually, when all bsp has its own config, I will remove the example's config later.

**Review by hathach — APPROVED** [review]
thank you, I have tested this with my frdm k64f board, working well. The save/restore crystal-less configuration is required, otherwise setting will be reset. There is no harm to restore these value when  not using crystal-less mode

**Comment by karlp** [comment]
well, my only other concern is that not all the kinetis parts have those crystalless config registers, so we're going to be writing into registers that don't always exist.

**Comment by hathach** [comment]
I will pull out to test with my kl25z board, I think it shouldn't have any effect. In my experience, these registers exist anyway, it is just readonly. If it is an issue, We can actually check the IRC_EN bit before restoring value, but I guess it is not really needed.

**Comment by hathach** [comment]
PS: it cause out-of-memory issue on KL25Z, PR is on the way to fix this :)

**Comment by karlp** [comment]
I'm back from holidays, and this asserts (`vApplicationStackOverflowHook 43: ASSERT FAILED
`) when a hub is plugged in, with the `examples/host/cdc_msc_hid_freertos` example.

My local version (didn't get to a PR) works fine.   I can't see what the difference would be (yet) but I'm obviously goin to have to find out :)  Did this example work for you at all?

Related to the FreeRTOSConfig files, the newly added file is not even used at present.  At least in the host/cdc_msc_hid_freertos.  I added #error calls, and it's still using the standard one from the example.  (however, the diff is minimal, and it shouldn't be changing stack sizes as far as I can tell)

**Comment by karlp** [comment]
You also removed the pinmuxing, are you sure this actually works with uarts after those changes?
